### PR TITLE
CORTX-32703 : Fetch original tag from latest image

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
# Problem Statement
- Need actual tag of latest tagged image

# Design
-  Fetching actual tag from image metadata.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing -
   1N - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8219/
   3N - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8221/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shailesh-vaidya — COMMENTED** [review]
@gauravchaudhari02 - Just one small change. Maybe we can take it as improvement also. 
Jenkins job still shows `latest` in status - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8168/ 
```
==================== Image Details ====================

ghcr.io/seagate/consul:1.11.4
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-latest
cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-latest
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-latest
ghcr.io/seagate/kafka:3.0.0-debian-10-r97
cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-latest
ghcr.io/seagate/zookeeper:3.8.0-debian-10-r9
```

File: solutions/kubernetes/functions.sh [hunk 1]
```diff
@@ -104,7 +104,11 @@ function pull_image() {
     local image="$1"
     if [[ "$image" =~ "latest" ]]; then
         docker pull "$image" &>/dev/null
-        actual_image_tag=$(docker inspect --format='{{ index .Config.Cmd}}' "$image" | grep -o "2.0.0-[0-9][0-9][0-9]*")
+        if [[ "$image" =~ "cortx-docker.colo.seagate.com" ]]; then
+            actual_image_tag=$(docker run --rm -t "$image" cat /opt/seagate/cortx/RELEASE.INFO | grep VERSION | awk -F'"' '{print $2}')
```
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply solutions/kubernetes/functions.sh]: Done

**Comment by gauravchaudhari02** [comment]
> @gauravchaudhari02 - Just one small change. Maybe we can take it as improvement also. Jenkins job still shows `latest` in status - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8168/
> 
> ```
> ==================== Image Details ====================
> 
> ghcr.io/seagate/consul:1.11.4
> cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-latest
> cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-latest
> cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-latest
> ghcr.io/seagate/kafka:3.0.0-debian-10-r97
> cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-latest
> ghcr.io/seagate/zookeeper:3.8.0-debian-10-r9
> ```

Fixed it

File: solutions/kubernetes/cortx-deploy-functions.sh [hunk 2]
```diff
@@ -217,11 +220,11 @@ function update_solution_config(){
 
 function add_image_info() {
     add_secondary_separator "Updating cortx images info in solution.yaml"   
-    update_image control-pod $CORTX_CONTROL_IMAGE
-    update_image data-pod $CORTX_DATA_IMAGE
-    update_image server-pod $CORTX_SERVER_IMAGE
-    update_image ha-pod $CORTX_CONTROL_IMAGE
-    update_image client-pod $CORTX_DATA_IMAGE
+    update_image control-pod "$CORTX_ACTUAL_CONTROL_IMAGE"
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/cortx-deploy-functions.sh:223]: Can't  we use `update_image control-pod "$CORTX_SERVER_IMAGE"` directly here

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Do we need additional variables CORTX_ACTUAL_SERVER_IMAGE etc.

File: solutions/kubernetes/cortx-deploy-functions.sh (hunk 2 above)
Comment (gauravchaudhari02) [reply solutions/kubernetes/cortx-deploy-functions.sh]: Done and tested

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**